src/D2L.CodeStyle.Analyzers/Immutability/ImmutableDefinitionChecker.cs

```diff
@@ -10,29 +9,23 @@ internal sealed class ImmutableDefinitionChecker {
 
 		internal delegate void DiagnosticSink( Diagnostic diagnostic );
 
-		private readonly SemanticModel m_model;
+		private readonly Compilation m_compilation;
 		private readonly DiagnosticSink m_diagnosticSink;
 		private readonly ImmutabilityContext m_context;
 
 		public ImmutableDefinitionChecker(
-			SemanticModel model,
+			Compilation compilation,

```

Comment:
This will take a bit of refactoring in your stuff.
`SemanticModel` is per-document, so we might need access to other models when we're dealing with a partial class.